monai/transforms/compose.py

```diff
@@ -358,3 +359,61 @@ def inverse(self, data):
         for o in reversed(applied_order):
             data = apply_transform(self.transforms[o].inverse, data, self.map_items, self.unpack_items, self.log_stats)
         return data
+
+
+class SomeOf(Compose):
+    """
+    ``SomeOf`` provides the ability to randomly sample a fixed number or
+    up to a fixed number of transforms from a list of callables.
+
+    If sampling up to the max_samples, each invocation of SomeOf uniformly samples a target number of transforms
+    for that invocation.
+
+    At each invocation the set of transforms to apply is uniformly sampled without replacement.
+
+    Args:
+        transforms: list of callables.
+        map_items: whether to apply transform to each item in the input `data` if `data` is a list or tuple.
+            Defaults to `True`.
+        unpack_items: whether to unpack input `data` with `*` as parameters for the callable function of transform.
+            Defaults to `False`.
+        log_stats: whether to log the detailed information of data and applied transform when error happened,
+            for NumPy array and PyTorch Tensor, log the data shape and value range,
+            for other metadata, log the values directly. Default to `False`.
+        min_samples: minimum number of transforms to sample. Defaults to `0`.
+        max_samples: maximum number of transforms to sample. Defaults to `3`.
+        fixed: whether to sample exactly `max_samples` transforms, or up to it. Defaults to `False`.
```

Comment:
The purpose of these 3 parameters is to allow you the choice to configure the number of transforms applied with each call.
If `fixed=True`, and `max_samples=3`, then everytime you invoke `__call__`, it will always sample 3 number of transforms to apply. In this case, `min_samples` is ignored.

If `fixed=False`, and `min_samples=0`, `max_samples=3` then every `__call__` will first choose a number $x \in [0,3]$, which is the number of transforms to apply. Then it will sample $x$ transforms to apply.  In this example, all 3 parameters are relevant.

Reply:
Yes, I know that. 
But if only one arg `n_transforms`, when `n_transforms=3`, it will always sample 3 number of transforms to apply. 
And when `n_transforms=(0, 3)`, will first choose a number from 0-3, which is the number of transforms to apply.
If I misunderstand, just correct me, thanks!

Reply:
Thanks for your comments. Sorry, I should've picked a different example to try explain the goal. It could be that I'm also misunderstanding your main point so feel free to ask further questions.

Consider this setup:
```
s = SomeOf((A(), B(), C(), D(), E()), min_num_transforms=2, max_num_transforms=4, fixed=False)
```

Here, `n_transforms = 5`,  and with each call,  want to sample $x \in \\{2,3,4\\}$ transforms.

So if we call s twice, we could hypothetically have:
```
s(first_input)  # First call.  It randomly chooses x=3, and the transforms [ C(), A(), E() ] to apply
s(second_input)  # Second call. It randomly chooses x=2, and the transforms [ D(), C() ] to apply
```

So rather than just being restricted to always sampling a fixed number of transforms to apply, you can configure it to sample from a range of subsets within a specified size range.  To simplify the maths expressions, let $n$ be `n_transforms`, and $x$ the number of transforms to sample.  Let's ignore permutations for this discussion. The usual implementation of SomeOf (e.g., in Albumentations) takes a fixed $x$ and chooses from ${n \choose x}$ different subsets of transforms to apply.

Let $s$ be `min_num_transforms`, and $t$ be `max_num_transforms`. The goal of these extra parameters is to allow users to increase the sampling space of subsets of transforms to $$\sum_{x=s}^t  {n \choose x}$$

Reply:
Why can't `n_transforms=(2, 4)` do the things you clarified above? Maybe you misunderstood what I mean, I would suggest `n_transforms` can both accept `int` or `sequence`. The type of `n_transforms` specified by the user will determine whether the num of the transforms will be fixed or random. So it means that when `n_transforms` is `int`, it will do something just like `fixed=True`, and when `n_transforms` is a sequence like `(2, 4)`, it will do something like `fixed=False`.
If I misunderstood, correct me, thanks!

Reply:
I misunderstood your point before, sorry about that.  I have changed the constructor to now accept a `num_transforms` which can be an int, or a tuple specifying the lower/upper bounds.